### PR TITLE
Fix import URL eating tons of memory on large file downloads

### PR DESCRIPTION
**Pull request by nirvn**
This PR introduces the use of a temporary file to progressively save data as a file is being downloaded during the import URL process. 

This should prevent QField from crashed caused by running out of memory when downloading large files in this process.

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 1428dc5c31ee11372b9dd8fa58287e57d3aaed48: [**arm64-android**](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5826-ae93ea3d1b3bf5db4e65d8dace2a85c5477fba6b-android-arm64.apk)
<details>
<summary>Other architectures</summary>

- [arm-neon-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5826-ae93ea3d1b3bf5db4e65d8dace2a85c5477fba6b-android-arm-neon.apk)
- [x64-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5826-ae93ea3d1b3bf5db4e65d8dace2a85c5477fba6b-android-x64.apk)
- [x86-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5826-ae93ea3d1b3bf5db4e65d8dace2a85c5477fba6b-android-x86.apk)
- [**all access arm64-android**](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5826-ae93ea3d1b3bf5db4e65d8dace2a85c5477fba6b-android-arm64.apk)
- [all access arm-neon-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5826-ae93ea3d1b3bf5db4e65d8dace2a85c5477fba6b-android-arm-neon.apk)
- [all access x64-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5826-ae93ea3d1b3bf5db4e65d8dace2a85c5477fba6b-android-x64.apk)
- [all access x86-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5826-ae93ea3d1b3bf5db4e65d8dace2a85c5477fba6b-android-x86.apk)
</details>

File: src/core/appinterface.cpp [hunk 1]
```diff
@@ -353,26 +360,41 @@ void AppInterface::importUrl( const QString &url )
...
 
-        emit importEnded( QFileInfo( filePath ).absolutePath() );
+        temporaryFile->deleteLater();
+        delete temporaryFile;
         return;
       }
     }
 
+    temporaryFile->deleteLater();
+    delete temporaryFile;
```
Comment (m-kuhn) [inline src/core/appinterface.cpp:396]: Does `deleteLater()` have any effect if it's deleted the line after?
Do we need manual memory management here, or could we just use a `QTemporaryFile temporaryFile;` on the stack instead?

**Comment by m-kuhn** [comment]
good move

**Review by suricactus — APPROVED** [review]
Good! We need to double check whether the QFC integration does the same.

File: src/core/appinterface.cpp [hunk 2]
```diff
@@ -286,15 +296,18 @@ void AppInterface::importUrl( const QString &url )
...
   QNetworkReply *reply = manager->get( request );
   connect( reply, &QNetworkReply::downloadProgress, this, [=]( int bytesReceived, int bytesTotal ) {
+    temporaryFile->write( reply->readAll() );
     if ( bytesTotal != 0 )
     {
       emit importProgress( static_cast<double>( bytesReceived ) / bytesTotal );
     }
   } );
 
   connect( reply, &QNetworkReply::finished, this, [=]() {
```
[the remark this answers is not in the document]
Comment (nirvn) [reply src/core/appinterface.cpp:306]: That would scope out at the end of the importUrl function and kill the object no?

[the remark this answers is not in the document]
Comment (m-kuhn) [reply src/core/appinterface.cpp:306]: There can only be one `unique_ptr` wich is moved into the lambda. But I admit my understanding of this is not 100% water proof.

[the remark this answers is not in the document]
Comment (nirvn) [reply src/core/appinterface.cpp:306]: Doesn't work.

**Review by m-kuhn — APPROVED** [review]

**Pull request merged**